sources.go

```diff
@@ -43,3 +44,14 @@ func Args() *Pipe {
 	}
 	return Echo(s.String())
 }
+
+// Get creates a pipe containing the output of retrieving the web resource
+// given by the supplied url.
+func Get(url string) *Pipe {
+	p := NewPipe()
+	res, err := http.Get(url)
+	if err != nil {
+		return p.WithError(err)
+	}
+	return p.WithReader(res.Body)
```

Comment:
An interesting question here is whether it makes sense (or is possible) to expose the HTTP status code to the user somehow. Usually if you're calling something like Get(), you care if the result is 404, for example. Any thoughts?

Reply:
I don't see how that's possible right now with the current design of `Pipe`. I don't just want to go around adding random methods to it that have no value for other types of sources, filters and sinks.

One sensible approach would be to more closely mirror the behaviour of `curl` and just return all of the output, headers, status line, body, the lot.

Reply:
Yes, that's an option. I think it's likely to be more useful if `Get()` simply returns the document, without headers. For example, if you're downloading an install script, you could `Get()` it, save it to a file, then `Exec()` it. You don't want to have to strip out headers and status info.

On the other hand, if the download fails altogether (404, for example), there's no point carrying on. We need some way to detect this. How about setting the error status on the pipe if the HTTP status indicates a problem?